### PR TITLE
[2018.3] Fix to cmd_batch

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Fixing cmd_batch to work correctly when called via salt-api.

### What issues does this PR fix or reference?
#48141 

### Previous Behavior
When running from the salt-api this would result in an exception, presumably because salt.utils.versions was unavailable.

### New Behavior
import salt.utils.versions inside cmd_batch.

### Tests written?
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by isbm** [comment]
@garethgreenaway how come? This module is imported already and is used all over the place.

**Review by isbm — CHANGES_REQUESTED** [review]
This is really weird. The module is imported and is used all over the place. And only that one function fails?? How come?

**Review by gtmanfred — APPROVED** [review]

**Comment by gtmanfred** [comment]
Probably because of multiprocessing

File: salt/client/__init__.py [hunk 1]
```diff
@@ -559,6 +559,7 @@ def cmd_batch(
             {'stewart': {...}}
         '''
         if 'expr_form' in kwargs:
+            import salt.utils.versions
```
Comment (rallytime) [inline salt/client/__init__.py]: Can you add a comment here about why this later import is needed so we don't accidentally remove it in the future?

**Review by rallytime — APPROVED** [review]
And a test to boot! Well done.

**Comment by rallytime** [comment]
@garethgreenaway That test is failing, however. Can you take a look?

File: salt/client/__init__.py [hunk 2]
```diff
@@ -559,6 +559,11 @@ def cmd_batch(
             {'stewart': {...}}
         '''
         if 'expr_form' in kwargs:
+            # We need to re-import salt.utils.versions here
+            # even though it has already been imported.
+            # when cmd_batch is called via the NetAPI
+            # the module is unavailable.
```
Comment (isbm) [inline salt/client/__init__.py]: :+1:

**Review by isbm — APPROVED** [review]
@garethgreenaway thanks for the explanation comment.

**Pull request merged**